l1-contracts/test/foundry/integration/GatewayTests.t.sol

```diff
@@ -164,6 +166,73 @@ contract GatewayTests is L1ContractDeployer, HyperchainDeployer, TokenDeployer,
         vm.stopBroadcast();
     }
 
+    function test_recoverFromFailedChainMigration() public {
+        gatewayScript.registerGateway();
+        gatewayScript.moveChainToGateway();
+
+        // Setup
+        IBridgehub bridgehub = IBridgehub(l1Script.getBridgehubProxyAddress());
+        IStateTransitionManager stm = IStateTransitionManager(l1Script.getSTM());
+        bytes32 assetId = bridgehub.stmAssetIdFromChainId(migratingChainId);
+        bytes memory transferData;
+
+        {
+            IZkSyncHyperchain chain = IZkSyncHyperchain(bridgehub.getHyperchain(migratingChainId));
+            bytes memory initialDiamondCut = l1Script.getInitialDiamondCutData();
+            bytes memory chainData = abi.encode(AdminFacet(address(chain)).prepareChainCommitment());
+            bytes memory stmData = abi.encode(address(1), msg.sender, stm.protocolVersion(), initialDiamondCut);
+            transferData = abi.encode(uint256(1), address(0), migratingChainId, stmData, chainData);
+        }
+
+        address chainAdmin = IZkSyncHyperchain(bridgehub.getHyperchain(migratingChainId)).getAdmin();
+        IL1AssetRouter assetRouter = bridgehub.sharedBridge();
+        bytes32 l2TxHash = keccak256("l2TxHash");
+        uint256 l2BatchNumber = 5;
+        uint256 l2MessageIndex = 0;
+        uint16 l2TxNumberInBatch = 0;
+        bytes32[] memory merkleProof = new bytes32[](1);
+        bytes32 txDataHash = keccak256(bytes.concat(bytes1(0x01), abi.encode(chainAdmin, assetId, transferData)));
+
+        // Mock Call for Msg Inclusion
+        vm.mockCall(
+            address(bridgehub),
+            abi.encodeWithSelector(
+                IBridgehub.proveL1ToL2TransactionStatus.selector,
+                migratingChainId,
+                l2TxHash,
+                l2BatchNumber,
+                l2MessageIndex,
+                l2TxNumberInBatch,
+                merkleProof,
+                TxStatus.Failure
+            ),
+            abi.encode(true)
+        );
+
+        // Set Deposit Happened
+        vm.startBroadcast(address(bridgeHub));
+        assetRouter.bridgehubConfirmL2Transaction({
```

Comment:
is it possible to remove this by using the same data as in` gatewayScript.moveChainToGateway();` ? Not needed, but would be nice

Reply:
i was trying to get it correct but was having trouble getting the exact values. will spend some time trying again

Reply:
ah ok, I think there are better things to do then if you tried already.